src/app/teams/teams.component.ts

```diff
@@ -0,0 +1,64 @@
+import {Component, OnInit} from '@angular/core';
+import {DataService} from '../data.service';
+import {ActivatedRoute, Router} from '@angular/router';
+import {Team} from '../model/Team';
+
+@Component({
+  selector: 'app-teams',
+  templateUrl: './teams.component.html',
+  styleUrls: ['./teams.component.css']
+})
+export class TeamsComponent implements OnInit {
+
+  constructor(private dataService: DataService,
+              private router: Router,
+              private route: ActivatedRoute) {
+  }
+
+  teamsList = new Array<Team>();
+  selectedTeam: Team = new Team();
+  selectedTeamCrest = 'none';
+
+  message = '';
+  backgroundImageURL: string;
+
+  activeCompetitionNo: number;
+
+
+  ngOnInit(): void {
+    this.processUrlParams();
+  }
+
+  processUrlParams(): void {
+    this.route.queryParams.subscribe(
+      (params) => {
+        this.activeCompetitionNo = params['league_id'];
+        this.backgroundImageURL = 'url(../assets/img/' + this.activeCompetitionNo + '_motive.jpg';
+        this.loadData();
+      });
+  }
+
+  loadData(): void {
+    this.message = '';
+    console.log(this.activeCompetitionNo);
+    this.dataService.getLeagueTeams(this.activeCompetitionNo).subscribe(
+      input => {
+        this.teamsList = input;
+        if (this.teamsList.length === 0) {
+          this.message = 'There is no data for this particular league.';
+        }
+      },
+      error => {
+        this.message = 'Error connecting to the server. Try again later.';
```

Comment:
Ja widzę dwie opcje - po pierwsze zamiast ustawiać różną wiadomość miej dwie wiadomości w staticu jakimś "No data" i "There was a problem contacting a server" a w HTMLu napisz 2 sekcje jedna zwykła pokazująca No data zamiast tabelki jeśli brak wyników z zapytania, 2 sekcja np. z czerwoną czcionką, że jest jakiś problem z serwerem, żeby wizualnie się różniło. Druga opcja wg mnie lepsza to jeżeli masz teamsList to pokaż tabelkę, jeśli nie to pokaż "No data" a w przypadku błędu wyświetl alert albo jakiś modal bootstrapowy z informacją o błędzie.